File: C#/Refactoring/refactoring_exercise_2/za/co/entelect/refactoring2/controller/ImageServiceDelegate.cs
using System;
using refactoring_exercise_2.za.co.entelect.refactoring2.domain;
using refactoring_exercise_2.za.co.entelect.refactoring2.service;

namespace refactoring_exercise_2.za.co.entelect.refactoring2.controller
{

    public class ImageServiceDelegate
    {

        private ImageService imageService = new ImageService();

        public Image Fetch(String id)
        {
            return imageService.Fetch(id);
        }

        public int Count()
        {
            return imageService.Count();
        }

        public void Add(Image image)
        {
            imageService.Add(image);
        }
    }
}

File: Java/refactoring_exercise_2/src/main/java/za/co/entelect/refactoring2/controller/ImageServiceDelegate.java
package za.co.entelect.refactoring2.controller;

import za.co.entelect.refactoring2.domain.Image;
import za.co.entelect.refactoring2.service.ImageService;


/**
 * Exercise 4:
 * This code ImageServiceDelegate suffers from the following code smell
 *
 * Middle Man
 */
public class ImageServiceDelegate {

   private ImageService imageService = new ImageService();

    public Image fetch(String id) {
        return imageService.fetch(id);
    }

    public int count() {
        return imageService.count();
    }

    public void add(Image image) {
        imageService.add(image);
    }
}
